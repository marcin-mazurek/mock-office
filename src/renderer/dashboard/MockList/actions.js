export const LOAD = 'mocks/LOAD';
export const UNLOAD = 'mocks/UNLOAD';

export const load = id => ({
  type: LOAD,
  id
});

export const unload = id => ({
  type: UNLOAD,
  id
});
