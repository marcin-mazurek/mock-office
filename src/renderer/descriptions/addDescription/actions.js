export const INIT = 'addDescription/INIT';
export const ADD = 'descriptions/ADD';

export const init = (serverId, descriptions) => ({
  type: INIT,
  serverId,
  descriptions
});

export const add = (serverId, descriptionId, descriptionPayload, title = 'untitled', interval, reuse, quantity, delay, requirements, blocking) => ({
  type: ADD,
  serverId,
  id: descriptionId,
  descriptionPayload,
  title,
  interval,
  reuse,
  quantity,
  delay,
  requirements,
  blocking
});
