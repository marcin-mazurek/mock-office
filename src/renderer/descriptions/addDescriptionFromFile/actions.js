export const INIT = 'pickDescriptionFile/INIT';

export const init = (serverId, files) => ({
  type: INIT,
  serverId,
  files
});
