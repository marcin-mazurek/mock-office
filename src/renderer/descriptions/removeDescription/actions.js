export const INIT = 'remove/init';
export const REMOVE = 'descriptions/REMOVE';

export const init = (serverId, descriptionId) => ({
  type: INIT,
  descriptionId,
  serverId
});

export const remove = (serverId, descriptionId) => ({
  type: REMOVE,
  serverId,
  descriptionId
});
