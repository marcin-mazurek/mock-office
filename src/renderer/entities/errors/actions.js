export const ADD = 'errors/ADD';

export const add = reason => ({
  type: ADD,
  reason
});
