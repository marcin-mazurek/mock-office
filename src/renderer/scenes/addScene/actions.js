export const INIT = 'addScene/INIT';
export const ADD = 'scenes/ADD';

export const init = (serverId, scenes) => ({
  type: INIT,
  serverId,
  scenes
});

export const add = (serverId, sceneId, scenePayload, title = 'untitled', interval, reuse, quantity, delay, requirements, blocking) => ({
  type: ADD,
  serverId,
  id: sceneId,
  scenePayload,
  title,
  interval,
  reuse,
  quantity,
  delay,
  requirements,
  blocking
});
