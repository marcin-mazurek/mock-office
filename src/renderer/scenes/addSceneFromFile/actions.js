export const INIT = 'pickSceneFile/INIT';

export const init = (serverId, files) => ({
  type: INIT,
  serverId,
  files
});
