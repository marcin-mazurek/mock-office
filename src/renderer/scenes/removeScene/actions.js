export const INIT = 'remove/init';
export const REMOVE = 'scenes/REMOVE';

export const init = (serverId, sceneId) => ({
  type: INIT,
  sceneId,
  serverId
});

export const remove = (serverId, sceneId) => ({
  type: REMOVE,
  serverId,
  sceneId
});
