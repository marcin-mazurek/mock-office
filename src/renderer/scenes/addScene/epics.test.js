import { Map, Set } from 'immutable';
import configureMockStore from 'redux-mock-store';
import 'rxjs';
import { createEpicMiddleware } from 'redux-observable';
import { init } from './actions';

const mockedElectron = {
  remote: {
    require() {
      return {
        default: {
          find() {
            return {
              addScene() {
                return 'scene id';
              }
            };
          }
        }
      };
    }
  }
};

jest.mock('electron', () => mockedElectron);
const addSceneEpic = require('./epics').default;

let store;
const epicMiddleware = createEpicMiddleware(addSceneEpic);
const mockStore = configureMockStore([epicMiddleware]);

describe('addSceneEpic', () => {
  test('success flow snapshot', () => {
    store = mockStore(new Map({
      servers: new Map({
        entities: new Map({
          'server id': {
            type: 'server type',
            name: 'server name',
            port: 3000,
            id: 'server id'
          }
        }),
        selected: 'server id',
        running: new Set()
      }),
      scenes: new Map()
    }));

    store.dispatch(init('serverId', [
      {
        title: 'scene title',
        requirements: {
          url: '/some-path'
        },
        scenePayload: 'scenePayload'
      }
    ]));
    expect(store.getScenes()).toMatchSnapshot();
  });
});
