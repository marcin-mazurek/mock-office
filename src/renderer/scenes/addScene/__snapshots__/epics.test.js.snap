// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addSceneEpic success flow snapshot 1`] = `
Array [
  Object {
    "scenes": Array [
      Object {
        "scenePayload": "scenePayload",
        "requirements": Object {
          "url": "/some-path",
        },
        "title": "scene title",
      },
    ],
    "serverId": "serverId",
    "type": "addScene/INIT",
  },
  Object {
    "blocking": undefined,
    "delay": undefined,
    "scenePayload": "scenePayload",
    "id": "scene id",
    "interval": undefined,
    "quantity": undefined,
    "requirements": Object {
      "url": "/some-path",
    },
    "reuse": undefined,
    "serverId": "serverId",
    "title": "scene title",
    "type": "scenes/ADD",
  },
]
`;
